pkg/migrations/sql/idempotent/base.sql

```diff
@@ -614,15 +634,51 @@ GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.delete_series_catalog_row(name, bigint[
 ---------------------------------------------------
 
 CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_metric_table_name_if_exists(
-        metric_name text)
-    RETURNS TABLE (id int, table_name name)
+        schema text, metric_name text)
+    RETURNS TABLE (id int, table_name name, table_schema name, series_table name, is_view boolean)
 AS $func$
-   SELECT id, table_name::name
-   FROM SCHEMA_CATALOG.metric m
-   WHERE m.metric_name = get_metric_table_name_if_exists.metric_name
+DECLARE
+    rows_found bigint;
+BEGIN
+    IF get_metric_table_name_if_exists.schema != '' THEN
+        RETURN QUERY SELECT m.id, m.table_name::name, m.table_schema::name, m.series_table::name, m.is_view
+        FROM SCHEMA_CATALOG.metric m
+        WHERE m.table_schema = get_metric_table_name_if_exists.schema
+        AND m.metric_name = get_metric_table_name_if_exists.metric_name;
+        RETURN;
+    END IF;
+    
+    SELECT count(*) 
+    INTO rows_found
+    FROM SCHEMA_CATALOG.metric m
+    WHERE m.table_schema = 'SCHEMA_DATA'
+    AND m.metric_name = get_metric_table_name_if_exists.metric_name;
+
+    IF rows_found = 1 THEN
+        RETURN QUERY SELECT m.id, m.table_name::name, m.table_schema::name, m.series_table::name, m.is_view
+        FROM SCHEMA_CATALOG.metric m
+        WHERE m.table_schema = 'SCHEMA_DATA'
+        AND m.metric_name = get_metric_table_name_if_exists.metric_name;
+        RETURN;
+    END IF;
```

Comment:
Slight optimization for the SCHEMA_DATA case we don't need to query for the count, instead we can use the special FOUND variable instead
```suggestion
        RETURN QUERY SELECT m.id, m.table_name::name, m.table_schema::name, m.series_table::name, m.is_view
        FROM SCHEMA_CATALOG.metric m
        WHERE m.table_schema = 'SCHEMA_DATA'
        AND m.metric_name = get_metric_table_name_if_exists.metric_name;
        
        IF FOUND THEN
             RETURN;
         END IF;
```